ivy/functional/frontends/numpy/creation_routines/numerical_ranges.py

```diff
@@ -156,3 +156,27 @@ def __init__(self):
 
 
 ogrid = OGrid()
+
+
+@handle_numpy_dtype
+@to_ivy_arrays_and_back
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    # Check if stop < start and swap values if necessary
+    flip = 0
+    if stop < start:
+        start, stop = stop, start
+        flip = 1
+    # Compute the common ratio
+    cr = ivy.log(stop / start) / (num - 1 if endpoint else num)
+    # Generate the sequence of values using the common ratio
+    if dtype is None:
+        dtype = start.dtype
```

Comment:
You should do `start, stop = promote_types_of_numpy_inputs(x1, x2)`
and then `dtype=start.dtype`